File: app/src/main/java/com/hover/stax/database/Converters.java
package com.hover.stax.database;

import androidx.room.TypeConverter;

import java.util.ArrayList;
import java.util.Arrays;

public class Converters {

    @TypeConverter
    public String fromArray(ArrayList<String> strings) {
        if (strings == null) return null;
        StringBuilder string = new StringBuilder();
        for (String s : strings) string.append(s).append(",");
        return string.toString();
    }

    @TypeConverter
    public ArrayList<String> toArray(String concatenatedStrings) {
        return concatenatedStrings != null ? new ArrayList<>(Arrays.asList(concatenatedStrings.split(","))) : new ArrayList<>();
    }
}

File: app/src/main/java/com/hover/stax/database/Converters.kt
package com.hover.stax.database

import androidx.room.TypeConverter

class Converters {

    @TypeConverter
    fun fromArray(strings: ArrayList<String>?): String? {
        if (strings == null) return null
        val string = StringBuilder()
        for (s in strings) string.append(s).append(",")
        return string.toString()
    }

    @TypeConverter
    fun toArray(concatenatedStrings: String?): ArrayList<String> {
        return if (concatenatedStrings != null) ArrayList(listOf(*concatenatedStrings.split(",").toTypedArray())) else ArrayList()
    }
}